### ISSUE TITLE
Add Visible Focus Styles to Buttons, Form Field and Links 

### ISSUE THREAD

**Issue by davidakennedy** [CMSgov/design-system#558]
## What
I'd like to propose a change to how we visually style focus states.

## Why
Currently, the design system relies on default browser focus styles in some cases. This can be inconsistent across browsers, and sometimes, not meet color contrast requirements. Also, our focus shadow doesn't meet contrast requirements on some backgrounds.

This change would bring more consistency to focus styles, across all elements: links, buttons and form fields.

The accompanying pull request is here: https://github.com/CMSgov/design-system/pull/510

I have also set up a demo page with all the elements that take a focus style in the design system here: https://davidakennedy.github.io/focus-styles-demo/

I plan to keep that up to date.

As far as the how on focus styles, I've taken an approach [modeled after the GDS design system](https://design-system.service.gov.uk/get-started/focus-states/).

It uses a mixture of box shadows and borders to provide the focus styles. It employs two colors, a new bright yellow: `#ffde11` and our base color.

In feedback on the pull request, the overwhelming consensus landed on the fact that a new focus color, combined with this method, would give us the most flexibility on both light and dark color schemes.

At this point, we need to:

- [ ] Do more browser testing to make sure anything wasn't missed when implementing the styles.
- [ ] Answer any implementation questions in the PR.
- [ ] Work out any bugs in the implementation.

Questions and feedback welcome!
